components/DarkmodeButton.js

```diff
@@ -0,0 +1,114 @@
+import useTheme from "@/contexts/theme";
+import styled, { keyframes, css } from "styled-components";
+import { useState } from "react";
+
+export default function DarkmodeButton() {
+  const { themeMode, lightTheme, darkTheme } = useTheme();
+  const [isToggled, setToggled] = useState(false);
+
+  const onChangeToDark = (event) => {
+    const darkModeStatus = event.currentTarget.checked;
+    setToggled(!isToggled);
+    if (darkModeStatus) {
+      darkTheme();
+    } else {
+      lightTheme();
+    }
+  };
+
+  return (
+    <>
+      <ToggleContainer>
+        <ToggleBorder>
+          <Toggle $isToggled={isToggled}>
+            <StyledInput
+              type="checkbox"
+              value=""
+              checked={themeMode === "dark"}
+              className=""
+              onChange={onChangeToDark}
+            />
+          </Toggle>
+        </ToggleBorder>
+      </ToggleContainer>
+    </>
```

Comment:
No fragment needed here:

```suggestion
      <ToggleContainer>
        <ToggleBorder>
          <Toggle $isToggled={isToggled}>
            <StyledInput
              type="checkbox"
              value=""
              checked={themeMode === "dark"}
              className=""
              onChange={onChangeToDark}
            />
          </Toggle>
        </ToggleBorder>
      </ToggleContainer>
```